src/main/java/com/ciandt/techgallery/service/impl/CronServiceImpl.java

```diff
@@ -66,13 +76,22 @@ public void sendEmailtoFollowers() {
     CronJob cronJob = new CronJob();
     cronJob.setName(Constants.CRON_MAIL_JOB);
     cronJob.setStartTimestamp(new Date());
+    
     try {
-      
-      List<TechnologyFollowers> techFollowers = technologyFollowersDao.findAll();
-      if (techFollowers != null && techFollowers.size() > 0) {
-        for (TechnologyFollowers technologyFollowers : techFollowers) {
-          Technology technology = technologyFollowers.getTechnology().get();
-          if (technologyFollowers.getFollowers().size() > 0) {
+      List<TechGalleryUser> followers = techGalleryUserDao.findAllFollowers();
+      if (followers != null && followers.size() > 0) {
+        for (TechGalleryUser follower : followers) {
+          // TO used in mustache template
+          TechGalleryActivitiesTO techGalleryActivitiesTo = new TechGalleryActivitiesTO();
+          techGalleryActivitiesTo.setTimestamp(new Date());
+          techGalleryActivitiesTo.setFollower(follower);
+          techGalleryActivitiesTo.setAppName(Constants.APP_NAME);
+          List<TechnologyActivitiesTO> techActivitiesToList =
+              new ArrayList<TechnologyActivitiesTO>();
+
+          for (String id : follower.getFollowedTechnologyIds()) {
+            Technology technology = technologyDao.findById(id);
+            // Find the last executed cronJob time
```

Comment:
@brunoliberal Take a look at this method...it is long and hard to understand. Also, you already defined that this block should be a method using this comment. Please refactor this part to findLastExecutedCronJob